src/services/expensesService.js

```diff
@@ -0,0 +1,54 @@
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
```

Comment:
The function `getAllExpenses` has a performance issue. You are applying multiple filters to the `expenses` array, which means the array is being iterated over multiple times. Instead of chaining filters, you can use a single filter with a combined condition to improve performance.